src/workspace.coffee

```diff
@@ -791,36 +799,65 @@ class Workspace extends Model
   # * `regex` {RegExp} to search with.
   # * `options` (optional) {Object} (default: {})
   #   * `paths` An {Array} of glob patterns to search within
+  #   * `onPathsSearched` (optional) {Function}
   # * `iterator` {Function} callback on each file found
   #
-  # Returns a `Promise`.
+  # Returns a *thenable* object with a `cancel()` method that will cancel all
+  # of the underlying searches that were started as part of this scan.
   scan: (regex, options={}, iterator) ->
```

Comment:
Looks like it returns this right: https://github.com/atom/atom/blob/3f03317b08647aaba5ef4c470af618a36ca1b76a/src/workspace.coffee#L874

Which is a `Promise`, am I missing something?